test/complex.jl

```diff
@@ -1188,6 +1188,36 @@ end
     @test tanh(atanh(complex(-1.0,-1.0))) == complex(-1.0,-1.0)
 end
 
+@testset "mod, issue #37376" begin
+    #Testing on zeros and ones
+    @test isequal(mod(0+0im,1),Complex(0,0))
+    @test isequal(mod(Complex(0,0),2),Complex(0,0))
+    @test isequal(mod(Complex(1,0),1),Complex(0,0))
+    @test isequal(mod(Complex(0,1),1),Complex(0,0))
+    @test isequal(mod(Complex(1,1),1),Complex(0,0))
+    #Testing on integer numerator and denominator
+    @test isequal(mod(Complex(5,6),2),Complex(1,0))
+    @test isequal(mod(Complex(11,17),7),Complex(4,3))
+    @test isequal(mod(Complex(10,12),5),Complex(0,2))
```

Comment:
Probably it's sufficient to just check one case for each type, since the underlying `mod` function is tested elsewhere — we just want to verify that complex numbers pass through to the real `mod`.